components/03-organisms/menu/breadcrumbs/yds-breadcrumbs.js

```diff
@@ -2,15 +2,229 @@ Drupal.behaviors.breadcrumbs = {
   attach(context) {
     // Selectors.
     const breadcrumbsWrapper = context.querySelector('.breadcrumbs__wrapper');
+    const breadcrumbs = context.querySelector('.breadcrumbs');
+    const breadcrumbsMenu = context.querySelector('.breadcrumbs__menu');
+    const breadcrumbsControls = context.querySelectorAll(
+      '.breadcrumbs__control',
+    );
+    const breadcrumbsLinks = context.querySelectorAll('.breadcrumbs__link');
     const breadcrumbsButton = context.querySelector('.breadcrumbs__button');
+    const breadcrumbsInner = context.querySelector('.breadcrumbs__inner');
+    const controlsWidth = context.querySelector(
+      '.breadcrumbs__control--left',
+    ).offsetWidth;
+    let scrollIndicatorDir;
 
-    // Function to add/remove breadcrumbs-overflow value.
-    // Used to show all breadcrumbs on mobile
+    /**
+     * getFirstVisible
+     * @description Get the first item that is visible (not overflown).
+     * @returns The value of the left edge of the first fully visible item plus
+     * the width of the controls so that things aren't visually hidden by the
+     * absolutely positioned elements.
+     */
+    function getFirstVisible() {
+      const breadcrumbsLeft = breadcrumbs.getBoundingClientRect().left;
+      const breadcrumbsItems =
+        breadcrumbs.querySelectorAll('.breadcrumbs__item');
+      const visibleItems = [];
+
+      breadcrumbsItems.forEach((item) => {
+        if (
+          item.getBoundingClientRect().right >
+          breadcrumbsLeft + controlsWidth
+        ) {
+          visibleItems.push(item);
+        }
+      });
+
+      return visibleItems[1].offsetLeft - controlsWidth;
+    }
+
+    /**
+     * getLastHidden
+     * @description Get the last item that is overflown (not visible).
+     * @returns The value of the left edge of the first partially hidden item
+     * minus the width of the controls so that things aren't visually hidden by
+     * the absolutely positioned elements.
+     */
+    function getLastHidden() {
+      const breadcrumbsLeft = breadcrumbs.getBoundingClientRect().left;
+      const breadcrumbsItems =
+        breadcrumbs.querySelectorAll('.breadcrumbs__item');
+      const hiddenItems = [];
+
+      breadcrumbsItems.forEach((item) => {
+        if (item.getBoundingClientRect().left < breadcrumbsLeft) {
+          hiddenItems.push(item);
+        }
+      });
+
+      return hiddenItems[hiddenItems.length - 1].offsetLeft - controlsWidth;
```

Comment:
I think the only way to prevent this would be to set some sort of "disabled" flag on the arrow until the "transition" is done (the transition being the time it takes for the arrow to disappear, if needed) ... I think that's adding a lot of complexity for not a lot of benefit. Even if the JS error throws, it doesn't really hurt anything.

I'll add a todo to the code, but I'm not sure this will ever be a priority.

Reply:
@ModulesUnraveled I hear ya and will approve because it's likely an edge case. That being said, a JS error this high up on a page may result in JS not firing properly further down the page for another component. Something to consider.